src/components/QuizResults.tsx

```diff
@@ -71,14 +79,21 @@ export const QuizResults = () => {
                     display: "flex",
                     justifyContent: "center"
                 }}>
+
                     <Link href="https://www.sas.com/en_us/curiosity/at-work.html">Learn more about how curiosity is valued in the workplace.</Link>
                 </Box>
                 <Box sx={{
                     mt: 4,
                     display: "flex",
                     justifyContent: "space-between"
                 }}>
-                    <Link href={"/curious"}>Retake Quiz</Link>
+
+                    {/*if loading is true, show loading circle, else show Retake Quiz link*/}
+                    {loading
+                    ? <CircularProgress/>
+                    : <Link onClick={loadingOnClick} href="/curious">Retake Quiz</Link>
```

Comment:
For something as short as `setLoading(true);` you can always do it inline if you like.

`loadingOnClick` would become `() => {setLoading(true);}`. I'm ok with it either way though.